File: map-service/src/main/java/com/example/map/repository/CircleMemberRepository.java
package com.example.map.repository;

import com.example.map.entity.CircleMember;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface CircleMemberRepository extends JpaRepository<CircleMember, Long> {

    List<CircleMember> findByUserId(Long userId);

    List<CircleMember> findByCircleId(Long circleId);

    boolean existsByCircleIdAndUserId(Long circleId, Long userId);
}

File: apps/map-service/src/circle/circle-member.repository.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { CircleMember } from './circle-member.entity';

@Injectable()
export class CircleMemberRepository {
  constructor(
    @InjectRepository(CircleMember)
    private readonly repo: Repository<CircleMember>,
  ) {}

  findByUserId(userId: number): Promise<CircleMember[]> {
    return this.repo.find({
      where: { userId },
      relations: ['circle', 'circle.members'],
    });
  }

  findByCircleId(circleId: number): Promise<CircleMember[]> {
    return this.repo.find({ where: { circle: { id: circleId } } });
  }

  save(member: CircleMember): Promise<CircleMember> {
    return this.repo.save(member);
  }
}
